src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -0,0 +1,230 @@
+        /// <typeparam name="T">The entry point's return type.</typeparam>
+        /// <param name="entryPoint">The entry point.</param>
+        /// <param name="parseResult">The result of parsing the command-line options.</param>
+        /// <param name="createSimulator">A function that creates an instance of the simulator to use.</param>
+        private static async Task DisplayEntryPointResult<T>(
+            IEntryPoint<T> entryPoint, ParseResult parseResult, Func<IOperationFactory> createSimulator)
+        {
+            var simulator = createSimulator();
+            try
+            {
+                var value = await entryPoint.Run(simulator, parseResult);
+                if (!(value is QVoid))
+                {
+                    Console.WriteLine(value);
+                }
+            }
+            finally
+            {
+                if (simulator is IDisposable disposable)
+                {
+                    disposable.Dispose();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns true if the alias is not already used by an entry point option.
+        /// </summary>
+        /// <typeparam name="T">The entry point's return type.</typeparam>
+        /// <param name="entryPoint">The entry point.</param>
+        /// <param name="alias">The alias to check.</param>
+        /// <returns>True if the alias is available for use by the driver.</returns>
+        private static bool IsAliasAvailable<T>(IEntryPoint<T> entryPoint, string alias) =>
+            !entryPoint.Options.SelectMany(option => option.RawAliases).Contains(alias);
+
+        /// <summary>
+        /// Returns the default value and displays a warning if the alias is shadowed by an entry point option, and
+        /// returns the original value otherwise.
+        /// </summary>
+        /// <typeparam name="TOption">The type of the option values.</typeparam>
+        /// <typeparam name="TEntryPoint">The entry point's return type.</typeparam>
+        /// <param name="entryPoint">The entry point.</param>
+        /// <param name="alias">The primary option alias corresponding to the value.</param>
+        /// <param name="value">The value of the option given on the command line.</param>
+        /// <param name="defaultValue">The default value for the option.</param>
+        /// <returns>The default value or the original value.</returns>
+        private static TOption DefaultIfShadowed<TOption, TEntryPoint>(
+            IEntryPoint<TEntryPoint> entryPoint, string alias, TOption value, TOption defaultValue)
+        {
+            if (IsAliasAvailable(entryPoint, alias))
+            {
+                return value;
+            }
+            else
+            {
+                var originalForeground = Console.ForegroundColor;
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.Error.WriteLine($"Warning: Option {alias} is overriden by an entry point parameter name.");
```

Comment:
Nice! I like the warning.
```suggestion
                Console.Error.WriteLine($"Warning: Option {alias} is overridden by an entry point parameter name.");
```